daemon/stats/collector_unix_test.go

```diff
@@ -0,0 +1,48 @@
+// +build !windows
+
+package stats // import "github.com/docker/docker/daemon/stats"
+
+import (
+	"bufio"
+	"os"
+	"testing"
+
+	"gotest.tools/v3/assert"
+)
+
+func TestCollector_cpuNanoSeconds(t *testing.T) {
+	tests := []struct {
+		name     string
+		filename string
+		expected uint64
+	}{
+		{
+			name:     "zero",
+			filename: "zeroProcStat",
+			expected: 0,
+		},
+		{
+			name:     "large",
+			filename: "largeProcStat",
+			expected: 242280125100000000,
+		},
+		{
+			name:     "currentMax",
+			filename: "maxProcStat",
+			expected: 18446744073700000000,
+		},
+	}
+
+	collector := Collector{bufReader: bufio.NewReaderSize(nil, 128)}
+
+	for _, tt := range tests {
+		f, err := os.Open("./testdata/" + tt.filename)
```

Comment:
With `io.Reader` we can store the test values as consts and use `strings.NewReader`

Reply:
Deleted test files and switched to NewReader